src/main/java/rx/subjects/ReplaySubject.java

```diff
@@ -115,13 +117,13 @@ public void call(SubjectObserver<T> o) {
                 try {
                     for (;;) {
                         int idx = o.<Integer>index();
-                        int sidx = state.index;
+                        int sidx = state.index.get();
```

Comment:
Does this still apply when `UnboundedReplayState` extends `AtomicInteger` directly? `state` is already a local variable [here](https://github.com/ReactiveX/RxJava/pull/3488/files#diff-21e6b6854376bf54cd8a90003d4994cbL92)